query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresBasedRequestHandler.java

```diff
@@ -470,13 +477,25 @@ Observable<Row> convert(ResultSet resultSet) throws SQLException {
       int columnCount = metaData.getColumnCount();
       if (columnCount > 0) {
         for (int c = 1; c <= columnCount; c++) {
-          String colVal = resultSet.getString(c);
-          if (colVal != null) {
-            builder.addColumn(
-                Value.newBuilder().setValueType(ValueType.STRING).setString(colVal).build());
+          int colType = metaData.getColumnType(c);
+          Value convertedColVal;
+          if (colType == Types.ARRAY) {
+            Array colVal = resultSet.getArray(c);
+            convertedColVal =
+                Value.newBuilder()
+                    .setValueType(ValueType.STRING)
+                    .setString(
+                        MAPPER.writeValueAsString(
+                            colVal != null ? colVal.getArray() : Collections.emptyList()))
```

Comment:
In case of pinot, for `Collections.emptyList()` , it returns `["null"]`. We can add defaults in schema.